ypp/excitons/excitons_WFs.F

```diff
@@ -239,8 +241,13 @@ subroutine exciton_WFs(Xk,BS_E_degs,iq)
          k_dot_r_e  = dot_product(r_cell_cc(:,ir),Xk%ptbz(ikbz,:)/alat(:))*2._SP*pi
          kp_dot_r_h = dot_product(r_hole,Xk%ptbz(ikpbz,:)/alat(:))*2._SP*pi
          !
-         WF1_ir=WF_symm(rindex(ir_hole),(/iv,ikp,isp,i_sp_pol/))
-         WF2_ir=WF_symm(rindex(ir     ),(/ic,ikibz,is,i_sp_pol/))
+         if(i_sp_pol == 2) then
```

Comment:
No need to have the two alternatives `i_sp_v` + `i_sp_c` and `i_sp_pol`
All the lines with `i_sp_pol` can be safely removed

Reply:
I tried that, but then it gave me segmentation fault in the soc case (which worked fine before). This is why I left the if else.

Reply:
> Because BSS_eh_table needs to be allocated of size `5`, and not `3+n_sp_pol`

Before touching and messing it up, isn't this already done in ./src/modules/mod_BS_solvers.F ?

`308      YAMBO_ALLOC(BSS_eh_table,(BS_H_dim,5))`

Reply:
This is done in yambo. But ypp uses a different allocation in ypp/excitons/excitons_read.F